src/Analyzers/CSharp/CodeFixes/AddBraces/CSharpAddBracesCodeFixProvider.cs

```diff
@@ -54,7 +49,7 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
                 editor.ReplaceNode(statement, (currentStatement, g) =>
                 {
                     var embeddedStatement = currentStatement.GetEmbeddedStatement();
-                    return currentStatement.ReplaceNode(embeddedStatement, SyntaxFactory.Block(embeddedStatement));
+                    return embeddedStatement is null ? currentStatement : currentStatement.ReplaceNode(embeddedStatement, SyntaxFactory.Block(embeddedStatement));
```

Comment:
If `embeddedStatement` is null, doesn't that mean there is something wrong?

Either the analyzer is reporting on a wrong (unexpected) node, or there is a missing handling of a syntax node in:

https://github.com/dotnet/roslyn/blob/b0cd71f82fbada617a7cb5354f52bbc478c1ff1d/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs#L252-L266

Reply:
i think i'm not confident of this here due to teh rewriting going upwards.  but i could